src/app/client/src/app/modules/public/components/signup/signup.component.html

```diff
@@ -1,34 +1,34 @@
-<sui-modal [isClosable]="true" [size]="'large'" (dismissed)="redirect()" #modal>
+<sui-modal [isClosable]="true" [size]="'large'" (dismissed)="redirect(); modal.deny()" #modal>
   <div class="header">
     <div class="ui grid">
       <div class="eleven wide column">
         <span class="ui header">{{resourceService?.frmelmnts?.scttl?.signup}}</span>
       </div>
-      <div class="one wide column actions">
-        <h5 class="ui basic icon circular button deny">
-          <i class="ui remove icon"></i>
-        </h5>
-      </div>
     </div>
   </div>
   <div class="content">
-    <form [formGroup]="signUpForm" autocomplete="off" id="signUpForm" class="ui form">
-      <!--  ng-if="!newUser.loader.showLoader && !newUser.error.showError " -->
-      <div class="  field">
+    <form *ngIf="!showLoader" [formGroup]="signUpForm" autocomplete="off" id="signUpForm" class="ui form">
+      <div class="field">
         <div class=" two fields">
-          <div class="required field ">
+          <div class="required field" [ngClass]="signUpForm.controls.userName.invalid && signUpForm.controls['userName'].errors.required && !signUpForm.controls['userName'].pristine || (signUpForm.controls['userName'].errors && signUpForm.controls['userName'].errors.pattern) ? 'error' : ''">
             <label>{{resourceService?.frmelmnts?.lbl?.username}}</label>
             <input type="text" formControlName="userName" placeholder="eg:user.name , min 5 character required">
+            <div class="margin-bottom-10 ui pointing red basic label below" *ngIf="signUpForm.controls.userName.invalid && signUpForm.controls['userName'].errors.required && !signUpForm.controls['userName'].pristine || (signUpForm.controls['userName'].errors && signUpForm.controls['userName'].errors.pattern)">
+              {{resourceService?.messages?.stmsg?.m0087}}
+            </div>
           </div>
-          <div class="required field">
+          <div class="required field" [ngClass]="signUpForm.controls.password.invalid && signUpForm.controls['password'].errors.required && !signUpForm.controls['password'].pristine ? 'error' : ''">
```

Comment:
Does this cover the error when password is consisting of only whitespaces (one or more)?